lib/vanagon/component/dsl.rb

```diff
@@ -424,13 +434,43 @@ def environment(*env) # rubocop:disable Metrics/AbcSize
       def add_preinstall_action(pkg_state, scripts)
         pkg_state = Array(pkg_state)
         scripts = Array(scripts)
-
-        if pkg_state.empty? || !(pkg_state - ["install", "upgrade"]).empty?
-          raise Vanagon::Error, "#{pkg_state} should be an array containing one or more of ['install', 'upgrade']"
-        end
+        check_pkg_state(pkg_state, __method__)
         @component.preinstall_actions << OpenStruct.new(:pkg_state => pkg_state, :scripts => scripts)
       end
 
+      # Adds trigger for scripts to be run on specified pkg_state.
+      #
+      # @param pkg_state [String] the state in which the scripts should execute. Can be
+      #   one or multiple of 'install' and 'upgrade'
```

Comment:
Yardoc suggestion: you can specify all of the allowed classes for a method argument in Yard. That'd look like this:

```
# @param pkg_state [String, Array] the package state during which the scripts should execute.
#     Accepts either a single string ("install" or "upgrade"), or an Array of Strings (["install", "upgrade"]).
```